src/RuleMatcher.cc

```diff
@@ -893,8 +902,8 @@ void RuleMatcher::FinishEndpoint(RuleEndpointState* state) {
 
     ExecPureRules(state, true);
 
-    loop_over_list(state->matched_by_patterns, i)
-        ExecRulePurely(state->matched_by_patterns[i], state->matched_text[i], state, true);
+    std::for_each(state->pattern_matches.begin(), state->pattern_matches.end(),
+                  [this, state](auto& m) { ExecRulePurely(m.rule, m.text, state, true); });
```

Comment:
This is my personal preference, so YMMV, but I do prefer a simple for-loop over `std::for_each` any time ...

Reply:
Yep, thanks for mentioning!